src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -740,7 +809,11 @@
            hostprivkey)))
 
     :else
-    (generate-master-ssl-files! settings certname ca-settings)))
+    (generate-master-ssl-files! settings certname ca-settings))
+  (log/debug (format "%s\n%s" (i18n/trs "initialize-master-ssl invoked. CA Settings: ") (ks/pprint-to-string ca-settings)))
+  ;; Generate a file containing serial numbers of the configured compiler master certificates, if any.
+  (generate-compile-master-serials (:compile-masters ca-settings) (:compile-master-serials-path ca-settings) (:signeddir ca-settings))
```

Comment:
You may want to just pass in `ca-settings`, or the subset of keys that you want, and destructure the map in the argument list for `generate-compile-master-serials` rather than use positional arguments.